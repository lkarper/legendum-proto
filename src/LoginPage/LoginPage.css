.LoginPage__section {
    display: flex;
    align-items: center;
    flex-direction: column;
    margin: 10px;
    padding: 15px;
    width: 100%;
    font-family: 'EB Garamond', serif;
}

.LoginPage__h2 {
    font-weight: 400;
    padding: 5px;
    border-bottom: 1px solid black;
    font-size: 24px;
}
