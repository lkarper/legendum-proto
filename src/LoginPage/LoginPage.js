import React, { useState, useContext } from 'react';
import UserContext from '../contexts/UserContext';
import NotesService from '../services/notes-service';
import ProgressService from '../services/progress-service';
import AuthApiService from '../services/auth-api-service';
import LoginForm from '../LoginForm/LoginForm';

const LoginPage = (props) => {

    const context = useContext(UserContext);

    const { forceUpdate } = props;

    const [userName, setUserName] = useState('');
    const [password, setPassword] = useState('');
    const [error, setError] = useState(null);

    const onLoginSuccess = () => {
        const { location, history } = props;
        const destination = (location.state || {}).from || '/dashboard';
        Promise.all([NotesService.getFetchNotesCallByUser(), ProgressService.getFetchProgressCallForUser()])
            .then(res => Promise.all(res.map(res => res.json())))
            .then(values => {
                const notes = values[0];
                const progress = values[1];
                context.setNotes(notes);
                context.setProgress(progress);
                forceUpdate();
                history.push(destination);
            })
            .catch(error => {
                context.setError(error.message);
            });
    }

    const handleLogin = (event) => {
        event.preventDefault();
        setError(null);
        AuthApiService.postLogin({
            user_name: userName,
            password
        })
            .then(res => {
                setUserName('');
                setPassword('');
                onLoginSuccess();
            })
            .catch(res => {
                setError(res.error);
            });
    }
 
    return (
        <LoginForm 
            userName={userName}
            password={password}
            setUserName={setUserName}
            setPassword={setPassword}
            handleLogin={handleLogin}
            error={error}
        />
    )

}

export default LoginPage;