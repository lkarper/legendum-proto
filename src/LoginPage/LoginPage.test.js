import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import LoginPage from './LoginPage';

describe('LoginPage component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<LoginPage />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<LoginPage />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
