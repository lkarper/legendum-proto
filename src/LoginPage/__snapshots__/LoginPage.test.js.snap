// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renders the UI as expected 1`] = `
<LoginPage
  forceUpdate={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  location={
    Object {
      "state": Object {
        "from": "/dashboard",
      },
    }
  }
>
  <LoginForm
    error=""
    handleLogin={[Function]}
    password=""
    setPassword={[Function]}
    setUserName={[Function]}
    userName=""
  >
    <div
      className="LoginForm__container"
    >
      <h2
        className="LoginForm__h2"
      >
        Login
      </h2>
      <form
        className="LoginForm__form"
        onSubmit={[Function]}
      >
        <div
          className="LoginForm__input-label-container"
        >
          <label
            htmlFor="user-name"
          >
            Username:
          </label>
          <input
            autoComplete="username"
            id="user-name"
            name="user-name"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="LoginForm__input-label-container"
        >
          <label
            htmlFor="password"
          >
            Password:
          </label>
          <input
            autoComplete="current-password"
            id="password"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
            value=""
          />
        </div>
        <div
          className="LoginForm__error-container"
          role="alert"
        />
        <button
          className="LoginForm__submit button"
          disabled={true}
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </LoginForm>
</LoginPage>
`;
